src/Processors/Formats/Impl/AvroRowInputFormat.cpp

```diff
@@ -554,7 +554,7 @@ AvroDeserializer::Action AvroDeserializer::createAction(const Block & header, co
     }
 }
 
-AvroDeserializer::AvroDeserializer(const Block & header, avro::ValidSchema schema, const FormatSettings & format_settings)
+AvroDeserializer::AvroDeserializer(const Block & header, avro::ValidSchema schema, bool allow_missing_fields)
```

Comment:
Acceptable, but FormatSetting allows to add more without changing function interface.

Reply:
The problem here is we cannot keep reference to FormatSettings. Proved by sanitizer ;( 
FormatSettings is too heavy to copy.
I can switch to copying FormatSettings::avro both for Avro and for AvroConfluent.
What do you think?

Reply:
You're right. Let's keep as is.

Reply:
Decided to go with allow_missing_fields.